src/js/test/rpc/server.test.ts

```diff
@@ -262,5 +262,77 @@ describe("Server", function () {
           });
       }
     );
+
+    it("should apply coprocessors to multiples record batches", function () {
+      const { spyGetHandles } = createStubs(sinonInstance);
+      // transform coprocessor function
+      const uppercase = (record) => ({
+        ...record,
+        value: record.value.map((char) => {
+          if (char > 97 && char < 122) {
+            return char - 32;
+          } else {
+            return char;
+          }
+        }),
+      });
+
+      const coprocessors = [1, 2, 3].map((id) =>
+        createHandle({
+          globalId: BigInt(id),
+          apply: (recordBatch) => {
+            const result = new Map();
+            const transformedRecord =
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              recordBatch.map(({ header, records }) => {
+                return {
+                  header,
+                  records: records.map(uppercase),
+                };
+              });
+            result.set("result", transformedRecord);
+            return result;
+          },
+        })
+      );
+
+      spyGetHandles.returns(coprocessors);
+
+      return Promise.all([
+        client.process_batch({
+          // send 101 record batches for apply 1,2,3 coprocessor
```

Comment:
This comment isn't clear for me. Can you rephrase?

Reply:
Yes, I can do